tests/lib/rules/new-ember-data-packages.js

```diff
@@ -0,0 +1,185 @@
+
+const ruleTester = new RuleTester();
+ruleTester.run('new-ember-data-packages', rule, {
+  valid: [
+    {
+      code: `import Model from '@ember-data/model';`,
+      parserOptions,
+    },
+    {
+      code: `import Model, { attr } from '@ember-data';
+
+        export default Model.extend({
+          name: attr('string')
+        });
+      `,
+      parserOptions,
+    },
+  ],
+
+  invalid: [
+    {
+      code: `import DS from 'ember-data';
+
+        export default DS.Model.extend({
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
+          type: 'ImportDeclaration',
+        },
+        {
+          message: "Use `import Model from '@ember-data/model';` instead of using DS.Model",
+          type: 'Identifier',
+        },
+      ],
+    },
+    {
+      code: `import DS, { Model } from 'ember-data';
+
+        export default Model.extend({
+          name: DS.attr('string')
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
+          type: 'ImportDeclaration',
+        },
+        {
+          message: "Use `import { attr } from '@ember-data/model';` instead of using DS.attr",
+          type: 'Identifier',
+        },
+      ],
+    },
+    {
+      code: `import DS from 'ember-data';
+        const { Model } = DS;
+
+        export default Model.extend({
+          name: DS.attr('string')
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
+          type: 'ImportDeclaration',
+        },
+        {
+          message: "Use `import Model from '@ember-data/model';` instead of using DS destructuring",
+          type: 'Property',
+        },
+        {
+          message: "Use `import { attr } from '@ember-data/model';` instead of using DS.attr",
+          type: 'Identifier',
+        },
+      ],
+    },
+    {
+      code: `import Model from 'ember-data/model';
+
+        export default Model.extend({
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
+          type: 'ImportDeclaration',
+        },
+      ],
+    },
+    {
+      code: `import Model from 'ember-data/model';
+        import attr  from 'ember-data/attr';
+
+        export default Model.extend({
+          name: attr('string')
+        });
+      `,
+      parserOptions,
+      errors: [
+        {
+          message:
+            'Imports from new @ember-data packages should be prefered over imports from ember-data',
```

Comment:
likely should drop use of "new" here as well

Although we seem to be redeclaring this string a lot, maybe we should hoist it up to a module level const.

Reply:
I've done that.